### PR TITLE
fix: added botd and visitorid to amplitude call

### PR DESCRIPTION
**Pull request by natapg**

### CODE REVIEW

File: src/helpers/amplitude.ts [hunk 1]
```diff
@@ -5,12 +5,15 @@ const amplitudeInit = () => {
   amplitude.getInstance().init(AMPLITUDE_API_KEY, undefined, { apiEndpoint: AMPLITUDE_API_ENDPOINT })
 }
 
-export const amplitudeLogEvent = (eventName: string, properties?: Record<string, string>) => {
+export const amplitudeLogEvent = (eventName: string, properties?: Record<string, string>, visitorId?: string) => {
   if (process.env.NODE_ENV !== 'development') {
     amplitudeInit()
+    if (visitorId) {
+      amplitude.setUserId(visitorId)
```
Comment (dcorso21) [inline src/helpers/amplitude.ts:12]: I don't think we want to set the visitorId as the userId. Rather, we want to store it as a property.

File: src/helpers/amplitude.ts [hunk 2]
```diff
@@ -5,12 +5,15 @@ const amplitudeInit = () => {
...
+    }
     amplitude.getInstance().logEvent(eventName, properties)
   } else {
     // eslint-disable-next-line no-console
-    console.log(`Amplitude event ${eventName} triggered with props:`, properties)
+    console.log(`Amplitude event ${eventName}, visitorId: ${visitorId}, triggered with props:`, properties)
```
Comment (dcorso21) [inline src/helpers/amplitude.ts:17]: I'm curious why we are logging out these events (I realize it was here before).

**Review by dcorso21 — CHANGES_REQUESTED** [review]
Main thing to address is the visitorId set as the userId IMO.

File: src/components/Layout/index.tsx [hunk 3]
```diff
@@ -56,8 +59,14 @@ export function LayoutTemplate({ children, notificationBar }: LayoutTemplateProp
   }, [isEuUser, userCountry])
 
   useEffect(() => {
-    amplitudeLogEvent('view marketing page', { route: pathname })
-  }, [pathname])
+    if (visitorData && visitorId) {
+      amplitudeLogEvent(
+        'view marketing page',
+        { route: pathname, is_detected: visitorData.products.botd.data.bot.result },
```
[the remark this answers is not in the document]
Comment (jackspirou) [reply src/components/Layout/index.tsx:65]: @dcorso21 my understanding is @natapg was worried about using using "bot" incase bots are looking for the term "bot".  @natapg I thought we agreed on a different term though like `isAutomated` or something.

File: src/helpers/amplitude.ts (hunk 2 above)
Comment (jackspirou) [reply src/helpers/amplitude.ts:17]: Yeah this is probably leftover from debugging

Comment (natapg) [reply src/helpers/amplitude.ts:17]: It is only for development (when `process.env.NODE_ENV === 'development'`)

File: src/helpers/amplitude.ts (hunk 1 above)
Comment (natapg) [reply src/helpers/amplitude.ts]: changed, please check @dcorso21

File: src/helpers/amplitude.ts [hunk 4]
```diff
@@ -5,12 +5,15 @@ const amplitudeInit = () => {
   amplitude.getInstance().init(AMPLITUDE_API_KEY, undefined, { apiEndpoint: AMPLITUDE_API_ENDPOINT })
 }
 
-export const amplitudeLogEvent = (eventName: string, properties?: Record<string, string>) => {
+export const amplitudeLogEvent = (eventName: string, properties?: Record<string, string>, visitorId?: string) => {
   if (process.env.NODE_ENV !== 'development') {
     amplitudeInit()
+    if (visitorId) {
+      amplitude.getInstance().identify(new amplitude.Identify().set('visitorId', visitorId))
```
Comment (dcorso21) [inline src/helpers/amplitude.ts:12]: After checking the docs, I see that this is valid. Optional: this is also valid: 

`amplitude.getInstance().setUserProperties({ visitorId });`

**Review by dcorso21 — APPROVED** [review]
Thanks, one optional change but looks good.

**Pull request merged**